### PR TITLE
Improvements in script test-conn

### PR DESCRIPTION
**Pull request by ramonhpr**
This PR verifies the error code in knot protocol to retry the schema and auth messages. Also, it fixes the missing `payload_len` field when knotd sends a message to the thing.

### CODE REVIEW

File: test/test-conn.py [hunk 1]
```diff
@@ -168,11 +168,12 @@ def send_knot_msg_auth(sock, uuid, token):
     return sock.send(msg)
```
Comment (vitbaq) [inline test/test-conn.py]: int the commit 1c41ee1fcb00d13b9a44b74766d418242d6f4756:

Send schema end if it is the last one in test-conn

File: test/test-conn.py [hunk 2]
```diff
@@ -210,7 +210,15 @@ def handle_auth():
         logging.info('Data_item %d will send every %d seconds', data['sensor_id'], interval)
```
Comment (vitbaq) [inline test/test-conn.py]: in the commit f41f7855d0e8e167d4d80bb88d01a08c05fb54c8:

Resend schema if receive an error in script test-conn

File: src/msg.c [hunk 3]
```diff
@@ -1081,7 +1081,7 @@ static bool handle_device_added(struct session *session, const char *device_id,
 	msg_credential_create(&msg, device_pending->uuid, token);
```
Comment (vitbaq) [inline src/msg.c]: in the commit 5fa54fdd8b99688ff1eae81c422860086102cbea
Fix payload_len when sending a schema and a register resp

File: test/test-conn.py [hunk 4]
```diff
@@ -202,7 +202,12 @@ def handle_register(msg):
     sensor['sock'] = s
```
Comment (vitbaq) [inline test/test-conn.py]: in the commit 386ff07958b32bb6aa994d1fc2bd5e7dd2263bed:

Resend auth message if receive an error in script test-conn

Comment (ramonhpr) [reply test/test-conn.py]: done

File: src/msg.c (hunk 3 above)
Comment (ramonhpr) [reply src/msg.c]: Done

File: test/test-conn.py (hunk 2 above)
Comment (ramonhpr) [reply test/test-conn.py]: done

File: test/test-conn.py (hunk 1 above)
Comment (ramonhpr) [reply test/test-conn.py]: done

**Pull request merged**